src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1678,6 +1682,102 @@ void EmitSingleCharLoop(RegexNode node, RegexNode? subsequent = null, bool emitL
                 // It's left pointing to the backtracking label for everything subsequent in the expression.
             }
 
+            void EmitSingleCharLazy(RegexNode node, RegexNode? subsequent = null, bool emitLengthChecksIfRequired = true)
+            {
+                // Emit the min iterations as a repeater.  Any failures here don't necessitate backtracking,
+                // as the lazy itself failed to match.
+                if (node.M > 0)
+                {
+                    EmitSingleCharFixedRepeater(node, emitLengthChecksIfRequired);
+                }
+
+                // If the whole thing was actually that repeater, we're done.
+                if (node.M == node.N)
```

Comment:
What happens if I specify a repeater and it has a max iteration number? Won't this mean that this won't be checked any longer? For example, pattern:
`a{2,5}?b` If I'm reading this correctly then that pattern would basically just check for min right?

Reply:
Or does this mean that we are checking for cases where we have only somehting like `a{2,5}?` and not a subsequent character, at which point the max iteration becomes irrelevant for a lazy loop?

Reply:
node.M == node.N means the min and the max are the same, so this would be the case for a{2,2}?, but a{2,5}? would have M==2 and N==5.  Did I misunderstand the question?

Reply:
Ok I have now tested it locally and think I understand what is happening. So what I was seeing after these changes is that pattern `a{2,3}?` was emitting code that was only checking for lower bound (meaning it was only ensuring that there were two a's) and I thought that was because of these changes, more specifically because of this if check. But now I think that this is not the case, and I believe that pattern is not hitting this code path to begin with because I suppose we are optimizing the pattern to instead be `a{2}` which is effectively equivalent as a lazy loop would have never tried looking for a max iteration when it already found a lower one that satisfied the pattern.

Reply:
Right. If you just have the expression `a{2,3}?`, with nothing after it, we'll optimize that into the equivalent of `(?>a{2,3}?), i.e. we'll make it atomic.  And then as it's atomic, once it matches two a's, it won't ever try to match more than that, so we proceed to optimize it to be the same as `a{2}`.  If you put something after it, e.g. `a{2,3}?[ab]`, then it won't be transformed in that way.